### PR TITLE
Suggestions for #8961

### PR DESCRIPTION
**Pull request by olix0r**
Signed-off-by: Oliver Gould <ver@buoyant.io>

### CODE REVIEW

File: policy-controller/k8s/index/src/index.rs [hunk 1]
```diff
@@ -294,7 +294,8 @@ impl kubert::index::IndexNamespacedResource<k8s::Pod> for Index {
         let meta = pod::Meta::from_metadata(pod.metadata);
         let probes = pod
             .spec
-            .map(|ref spec| pod::get_http_probes(spec, &port_names))
+            .as_ref()
+            .map(pod::pod_http_probes)
```
Comment (olix0r) [inline policy-controller/k8s/index/src/index.rs:298]: `port_names` is unused

File: policy-controller/k8s/index/src/pod.rs [hunk 2]
```diff
@@ -72,48 +72,44 @@ pub(crate) fn port_names(spec: &Option<k8s::PodSpec>) -> HashMap<String, PortSet
...
-                    }
+pub(crate) fn pod_http_probes(pod: &k8s::PodSpec) -> PortMap<HashSet<String>> {
+    let mut probes = PortMap::<HashSet<String>>::default();
+    for (port, path) in pod.containers.iter().flat_map(container_http_probe_paths) {
+        probes.entry(port).or_default().insert(path);
+    }
+    probes
+}
+
+fn container_http_probe_paths(container: &k8s::Container) -> Vec<(NonZeroU16, String)> {
```
Comment (olix0r) [inline policy-controller/k8s/index/src/pod.rs:83]: I think this benefits from being split into some smaller functions. This lets us use `?` short-circuiting.

Also, it seems simpler to collect all the port-paths and then join them into a set keyed on port.

File: policy-controller/k8s/index/src/pod.rs [hunk 3]
```diff
@@ -72,48 +72,44 @@ pub(crate) fn port_names(spec: &Option<k8s::PodSpec>) -> HashMap<String, PortSet
...
+    fn find_by_name(name: &str, ports: &[k8s::ContainerPort]) -> Option<NonZeroU16> {
+        for port in ports {
+            if let Some(ref n) = port.name {
+                if n.eq_ignore_ascii_case(name) {
```
Comment (olix0r) [inline policy-controller/k8s/index/src/pod.rs:87]: seems like we should do case-agnostic port name comparison

File: policy-controller/k8s/index/src/pod.rs [hunk 4]
```diff
@@ -72,48 +72,44 @@ pub(crate) fn port_names(spec: &Option<k8s::PodSpec>) -> HashMap<String, PortSet
...
     }
-    http_probes
+
+    (container.liveness_probe.iter())
+        .chain(container.readiness_probe.iter())
+        .chain(container.startup_probe.iter())
+        .filter_map(|p| {
+            let probe = p.http_get.as_ref()?;
+            let port = get_port(&probe.port, container)?;
+            let path = probe.path.clone().unwrap_or_else(|| "/".to_string());
```
Comment (olix0r) [inline policy-controller/k8s/index/src/pod.rs:109]: As mentioned in #8961, I don't think we should panic if a path is unset.

**Review by kleimkuhler — APPROVED** [review]
Looks good. I like the handling of `fn get_port` returning an `Option`

**Pull request merged**